cores/esp8266/core_esp8266_wiring.cpp

```diff
@@ -56,6 +57,22 @@ void delay(unsigned long ms) {
     }
 }
 
+void delay(unsigned long ms) {
+    if (ms) {
+        using esp8266::polledTimeout::oneShotMs;
+        oneShotMs waitDone(ms);
+        while(!waitDone) {
+            os_timer_setfn(&delay_timer, (os_timer_func_t*) &delay_end, 0);
+            os_timer_arm(&delay_timer, 1, ONCE);
```

Comment:
To reduce power consumption during delay cycles i've left os_timer_arm and no used something like
```cpp
using oneShotYieldMs = polledTimeout::timeoutTemplate<false, YieldPolicy::YieldOrSkip>;
oneShotYieldMs waitDone(ms);
while (!waitDone);
```